File: src/main/java/com/example/librarymanagement/dto/mapper/AuthorMapper.java
package com.example.librarymanagement.dto.mapper;

import com.example.librarymanagement.dto.AuthorDto;
import com.example.librarymanagement.model.entity.Author;
import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

@Component
public class AuthorMapper {
    public Author toAuthor(AuthorDto authorDto) {
        Author author = new Author();
        author.setFirstName(authorDto.getFirstName());
        author.setLastName(authorDto.getLastName());
        return author;
    }

    public AuthorDto toAuthorDto(Author author) {
        AuthorDto authorDto = new AuthorDto();
        authorDto.setId(author.getId());
        authorDto.setFirstName(author.getFirstName());
        authorDto.setLastName(author.getLastName());
        return authorDto;
    }

    public List<AuthorDto> toAuthorDto(List<Author> authors) {
        if (authors == null || authors.isEmpty()) return new ArrayList<>();

        return authors.stream()
                .map(this::toAuthorDto)
                .toList();
    }
}

File: src/main/java/com/example/librarymanagement/dto/mapper/AuthorMapper.kt
package com.example.librarymanagement.dto.mapper

import com.example.librarymanagement.dto.AuthorDto
import com.example.librarymanagement.model.domain.Author
import org.springframework.stereotype.Component

@Component
class AuthorMapper {
    fun toAuthor(authorDto: AuthorDto, id: String? = null): Author {
        return Author(
            id = id,
            firstName = authorDto.firstName,
            lastName = authorDto.lastName )
    }

    fun toAuthorDto(author: Author): AuthorDto = AuthorDto(author.id!!, author.firstName, author.lastName)

    fun toAuthorDto(authors: List<Author>): List<AuthorDto> = authors.map { toAuthorDto(it) }
}
